onedal/objective_function/objective_function.py

```diff
@@ -0,0 +1,307 @@
+
+        X_table, coef_table = to_table(X_loc, coef_loc)
+        y_table = to_table(y_loc)
+
+        result = self.func(policy, params, X_table, coef_table, y_table)
+
+        options = self._get_result_options(options)
+        options = options.split("|")
+
+        res = {opt: getattr(result, opt) for opt in options}
+
+        n = X.shape[0]
+
+        return {k: from_table(v).ravel() * n for k, v in res.items()}
+
+
+class LogisticLoss(BaseObjectiveFunction):
+    def __init__(
+        self, *, algorithm="by_default", fit_intercept=True, **kwargs
+    ):
+        self.fit_intercept = fit_intercept
+        super().__init__(algorithm, _backend.objective_function.compute.logloss)
+
+    def change_gradient_format(self, grad, coef, l2_reg_strength):
+        """Change the gradient layout
+
+        onedal compute_gradient function returns:
+        [dL / dw_0, dL / dw_1, ...., dL / dw_p] if fit_intercept=True
+        [0.0, dL / dw_1, ...., dL / dw_p] if fit_intercept=False
+        gradient function from sklearn.linear_model._linear_loss returns:
+        [dL / dw_1, ...., dL / dw_p, dL / dw_0] if fit_intercept=True
+        [dL / dw_1, ...., dL / dw_p] if fit_intercept=False
+
+        so to align with python interface format should be changed
+        """
+
+        if self.fit_intercept:
+            grad = np.hstack([grad[1:] + coef[:-1] * l2_reg_strength, grad[0]])
+        else:
+            grad = grad[1:] + coef * l2_reg_strength
+        return grad
+
+    def change_hessian_format(self, hess, coef, l2_reg_strength):
+        """Change the hessian layout
+        onedal compute_hessian function returns matrix H of size (p+1)*(p+1)
+        H_i,j = dL / (dw_i * d_w_j)
+        if fit_intercept=False: H_0,i = H_0,i = 0.0
+
+        hessian function from sklearn.linear_model._linear_loss returns:
+
+        if fit_intercept=True
+        matrix of size (p + 1) * (p + 1)
+        H_i,j = dL / (dw_(i+1) * dw_(j+1)) for 0 <= i,j < p
+        H_i,p = H_p,i = dL / (dw_(i+1) dw_0)
+
+        if fit_intercept=False
+        matrix of size p * p
+        H_i,j = dL / (dw_(i+1) * dw_(j+1))
+
+        so to align with python interface format should be changed
+        """
+        num_params = coef.shape[0]
+        if not self.fit_intercept:
+            num_params += 1
+        hess = hess.reshape(num_params, num_params)
+        if self.fit_intercept:
+            hess = np.hstack(
+                (
+                    np.vstack(
+                        [hess[1:, 1:] + np.diag([l2_reg_strength] * (num_params - 1)),
+                         hess[0, 1:],]
+                    ),
+                    np.hstack([hess[0, 1:], hess[0][0]]).reshape(-1, 1),
+                )
+            )
+        else:
+            hess = hess[1:, 1:] + np.diag([l2_reg_strength] * (num_params - 1))
+        return hess
+
+    def __calculate_regularization(self, coef, l2_reg_strength):
```

Comment:
What about L1 regularization?

Reply:
There is no l1-regularization in LinearModelLoss class and the interface should align with it.